url53f943c5cb_Praznat_AGP_tgz-pJ8-AMath_TestingJ8/src/AMath/Calc.java

```diff
@@ -0,0 +1,506 @@
+			spin = Calc.squeezeByte(rand[i], 0, cumProps[cumProps.length-1]);
+			answer[i] = (byte) findLessThan(spin, cumProps);
+		}
+		return answer;
+	}
+	
+	public static void printArray(byte[] V) {
+		for(int i = 0; i < V.length; i++) {System.out.println(V[i]);}
+	}
+	public static void printArrayH(byte[] V) {
+		for(int i = 0; i < V.length; i++) {System.out.print(V[i] + "	");}
+		System.out.println();
+	}
+	public static void printArray(int[] V) {
+		for(int i = 0; i < V.length; i++) {System.out.println(V[i]);}
+	}
+	public static void printArrayH(int[] V) {
+		for(int i = 0; i < V.length; i++) {System.out.print(V[i] + "	");}
+		System.out.println();
+	}
+	public static void printArray(byte[][] M) {
+		for(int i = 0; i < M.length; i++) { for(int j = 0; j < M[i].length; j++) {
+			System.out.print(M[i][j] + "	");
+		}System.out.println();}
+	}
+	public static void printArray(String[][] M) {
+		for(int i = 0; i < M.length; i++) { for(int j = 0; j < M[i].length; j++) {
+			System.out.print(M[i][j] + "	");
+		}System.out.println();}
+	}
+	public static void printArray(int[][] M) {
+		for(int i = 0; i < M.length; i++) { for(int j = 0; j < M[i].length; j++) {
+			System.out.print(M[i][j] + "	");
+		}System.out.println();}
+	}
+	
+	
+
+    public static byte[][] transpose (byte[][] inM) {
+    	byte[][] outM = new byte[inM[0].length][];  //assumption that all vectors are equal length
+    	for (int i = 0; i < outM.length; i++) {
+    		outM[i] = new byte[inM.length];
+    		for (int j = 0; j < outM[i].length; j++) {
+    			outM[i][j] = inM[j][i];
+    		}
+    	}
+    	return outM;
+    }
+    public static int[][] transpose (int[][] inM) {
+    	int[][] outM = new int[inM[0].length][];  //assumption that all vectors are equal length
+    	for (int i = 0; i < outM.length; i++) {
+    		outM[i] = new int[inM.length];
+    		for (int j = 0; j < outM[i].length; j++) {
+    			outM[i][j] = inM[j][i];
+    		}
+    	}
+    	return outM;
+    }
+    public static String[][] transpose (String[][] inM) {
+    	String[][] outM = new String[inM[0].length][];  //assumption that all vectors are equal length
+    	for (int i = 0; i < outM.length; i++) {
+    		outM[i] = new String[inM.length];
+    		for (int j = 0; j < outM[i].length; j++) {
+    			outM[i][j] = inM[j][i];
+    		}
+    	}
+    	return outM;
+    }
+
+    public static int[] normalOrder(int L) {
+    	int[] V = new int[L];
+    	for (int i = 0; i < L; i++) {V[i] = i;}
+    	return V;
+    }
+    public static int[] randomOrder(int L) {
+    	int[] V = new int[L];
+    	for (int i = 0; i < L; i++) {V[i] = i;}
+    	for (int i = V.length - 1; i > 0; i--) {
+    		int j = AGPmain.rand.nextInt(i);
+    		int tmp = V[j];
+    		V[j] = V[i];
+    		V[i] = tmp;
+    	}
+    	return V;
+    }
+    public static void shuffle(int[] pop) {
+    	for (int i = pop.length - 1; i > 0; i--) {
+    		int j = AGPmain.rand.nextInt(i);
+    		int tmp = pop[j];
+    		pop[j] = pop[i];
+    		pop[i] = tmp;
+    	}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 365:370, 373:378.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.